.github/workflows/build-artifact-wrapper.yaml

```diff
@@ -0,0 +1,71 @@
+name: "Build tt-metal artifacts"
+
+on:
+  workflow_call:
+    inputs:
+      arch:
+        required: false
+        type: string
+        default: '["grayskull", "wormhole_b0"]'
+      build-type:
+        required: false
+        type: string
+        default: "Release"
+      tracy:
+        required: false
+        type: boolean
+        default: false
+        description: "Build with tracy enabled"
+      os:
+        required: false
+        type: string
+        default: "ubuntu-20.04-amd64"
+  workflow_dispatch:
+    inputs:
+      arch:
+        required: false
+        type: string
+        default: '["grayskull", "wormhole_b0"]'
+      build-type:
+        required: false
+        type: choice
+        options:
+          - Release
+          - Debug
+          - RelWithDebInfo
+          - CI
+      tracy:
+        required: false
+        type: boolean
+        default: false
+        description: "Build with tracy enabled"
+      os:
+        required: false
+        type: choice
+        default: "ubuntu-20.04-amd64"
+        options:
+          - "ubuntu-20.04-amd64"
+          - "ubuntu-22.04-amd64"
+
+permissions:
+  actions: read
+  contents: read
+  pages: write
+  id-token: write
```

Comment:
Does this need `id-token: write`?

Reply:
Borrowed permissions list from 
https://docs.github.com/en/actions/use-cases-and-examples/publishing-packages/publishing-docker-images
for publishing docker images. I got the permissions list from there

Reply:
the id-token is also for slack notifications if a step fails on main branch iirc*